src/codegen/compile-type-definition.lisp

```diff
@@ -19,6 +24,58 @@
                ,(constructor-entry-name constructor)
                (coalton-impl/codegen::F1 #',(constructor-entry-name constructor))))))
 
+      ((eql :development *interaction-mode*)
+       `((defclass ,(type-definition-name def) ()
+           ())
+
+         ,@(loop
+             :for constructor :in (type-definition-constructors def)
+             :for classname := (constructor-entry-classname constructor)
+             :for slot-types := (mapcar #'coalton-impl/typechecker::fresh-inst (constructor-entry-arguments constructor))
+             :for slot-names := (ctor-make-slot-names (length slot-types) package)
+             :append
```

Comment:
minor, but I prefer

```
:append
(list `(...))
```

To me, the backtick is a signifier of "syntax" (aka: "im writing a backtick because i want to quote syntax"), and which it's also used to construct a list, it can be confusing, even if equivalent

Reply:
The pattern of ``    `(( `` is used throughout Coalton already, and `(list` adds more indentation.

Also isn't a list of quoted forms "quote syntax"?

Reply:
Ideally

    `(foo ...)

implies that

    (foo ...)

is a valid form (i.e., evaluatable expression, or sub-expression of an evaluatable expression).

    `((foo ...))

rarely falls under this class.

This is a purely stylistic point (like `':quoted-keywords`, two-branch-`if`, `cond`-for-effects, etc).

Reply:
That makes sense. I can update this file.